### PR TITLE
Split GitHub workflows out into build and release

### PR DESCRIPTION
**Pull request by jwthomson**
The changes in this PR are based loosely on the changes made to enable similar functionality in the speedometer repo: https://github.com/CompositionalIT/Feliz-ReactSpeedometer

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -0,0 +1,30 @@
+name: Build
+
+on:
+  push:
```
[the remark this answers is not in the document]
Comment (jwthomson) [reply .github/workflows/build.yml:4]: I can't see an alternative which would seem to include merges. I would have thought that a merge is just a merge commit, which is then pushed, and then the push would trigger this workflow. Or perhaps my understanding is lacking?

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -1,36 +1,39 @@
...
-        dotnet-version: 6.0.300
+
     - name: Restore tools
       run: dotnet tool restore
+
     - name: Restore dependencies
       run: dotnet restore
       working-directory: ./src
+
     - name: Build
```
[the remark this answers is not in the document]
Comment (jwthomson) [reply .github/workflows/release.yml:29]: Good idea. Let's make this an investigation task for another day!

File: .github/workflows/build.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (mattgallagher92) [reply .github/workflows/build.yml:4]: A merge wouldn't have to be pushed if it was done directly within the GitHub repo

[the remark this answers is not in the document]
Comment (mattgallagher92) [reply .github/workflows/build.yml:4]: This does trigger on merge to main.

**Pull request merged**